File: Back-End/src/main/java/iuh/fit/ecommerce/services/CategoryBrandService.java
package iuh.fit.ecommerce.services;

import iuh.fit.ecommerce.dtos.request.categoryBrand.SetBrandsForCategoryRequest; // <-- THÊM
import iuh.fit.ecommerce.dtos.response.base.ResponseWithPagination;
import iuh.fit.ecommerce.dtos.response.brand.BrandResponse;
import iuh.fit.ecommerce.dtos.response.category.CategoryResponse; // Giả định bạn có DTO này
// import iuh.fit.ecommerce.dtos.response.categoryBrand.CategoryBrandResponse; // <-- XÓA
import java.util.List;

public interface CategoryBrandService {

    void setBrandsForCategory(SetBrandsForCategoryRequest request);


    List<BrandResponse> getBrandsByCategoryId(
            Long categoryId, String brandName
    );

    List<CategoryResponse> getCategoriesByBrandId(
            Long brandId, String categoryName
    );

    List<BrandResponse> getBrandsByCategorySlug(String slug);
}

File: Front-End/src/services/categoryBrand.service.ts
import axiosClient from '@/configurations/axios.config';
import type {
  SetBrandsForCategoryRequest,
  BrandListByCategoryResponse,
  CategoryListByBrandResponse,
} from '@/types/category-brand.type';
import type { ResponseApi } from '@/types/responseApi.type';
import type { Brand } from '@/types/brand.type';

export const categoryBrandService = {
  getBrandsByCategoryId: async (
    categoryId: number,
    brandName: string = ""
  ) => {
    const query = brandName
      ? `?brandName=${encodeURIComponent(brandName)}`
      : '';
    const response = await axiosClient.get<BrandListByCategoryResponse>(
      `/category-brands/categories/${categoryId}/brands${query}`
    );
    return response.data;
  },

  getCategoriesByBrandId: async (
    brandId: number,
    categoryName: string = ""
  ) => {
    const query = categoryName
      ? `?categoryName=${encodeURIComponent(categoryName)}`
      : '';
    const response = await axiosClient.get<CategoryListByBrandResponse>(
      `/category-brands/brands/${brandId}/categories${query}`
    );
    return response.data;
  },

  setBrandsForCategory: async (request: SetBrandsForCategoryRequest) => {
    const response = await axiosClient.post<ResponseApi<void>>(
      '/category-brands/set-brands',
      request
    );
    return response.data;
  },

  getBrandsByCategorySlug: async (slug: string) => {
    const response = await axiosClient.get<ResponseApi<Brand[]>>(
      `/category-brands/categories/slug/${slug}/brands`
    );
    return response.data;
  },
};